flagsmith/resource_tag.go

```diff
@@ -0,0 +1,212 @@
+)
+
+// Ensure provider defined types fully satisfy framework interfaces
+var _ resource.Resource = &tagResource{}
+var _ resource.ResourceWithImportState = &tagResource{}
+
+func newTagResource() resource.Resource {
+	return &tagResource{}
+}
+
+type tagResource struct {
+	client *flagsmithapi.Client
+}
+
+func (r *tagResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_tag"
+}
+
+func (r *tagResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*flagsmithapi.Client)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *flagsmithapi.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+
+	r.client = client
+}
+
+func (t *tagResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "Flagsmith Tag",
+
+		Attributes: map[string]schema.Attribute{
+			"id": schema.Int64Attribute{
+				Computed:            true,
+				MarkdownDescription: "ID of the tag",
+				PlanModifiers:       []planmodifier.Int64{int64planmodifier.UseStateForUnknown()},
+			},
+			"uuid": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "UUID of the tag",
+				PlanModifiers:       []planmodifier.String{stringplanmodifier.UseStateForUnknown()},
+			},
+			"project_id": schema.Int64Attribute{
+				Computed:            true,
+				MarkdownDescription: "ID of the project",
+				PlanModifiers:       []planmodifier.Int64{int64planmodifier.UseStateForUnknown()},
+			},
+			"tag_name": schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "Name of the tag",
+			},
+			"tag_colour": schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "Hexadecimal value of the tag color",
+			},
+			"description": schema.StringAttribute{
+				Optional:            true,
+				MarkdownDescription: "Description of the feature",
+			},
+			"project_uuid": schema.StringAttribute{
+				MarkdownDescription: "UUID of project the tag belongs to",
+				Required:            true,
+				PlanModifiers:       []planmodifier.String{stringplanmodifier.RequiresReplace()},
+			},
+		},
+	}
+}
+
+func (r *tagResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data TagResourceData
+
+	diags := req.Config.Get(ctx, &data)
+	resp.Diagnostics.Append(diags...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	clientTag := data.ToClientTag()
+
+	// Create the feature
```

Comment:
This comment is wrong (it should say `Create the tag`) but tbh I'd suggest we just remove this comment, it seems unnecessary to me. 

```suggestion
```